packages/server/src/unstable-core-do-not-import/http/resolveResponse.ts

```diff
@@ -371,9 +379,27 @@ export async function resolveResponse<TRouter extends AnyRouter>(
 
     // ----------- response handlers -----------
     if (!info.isBatchCall) {
-      const [call] = info.calls;
+      const call = info.calls[0]!;
       const [error, result] = await rpcCalls[0]!;
 
+      if (!error && result?.data instanceof Response) {
```

Comment:
So my sense is that we should follow the patterns established by content types for inputs. It would be more consistent and more http-native

1. Have multiple response strategies, JsonResponseType, BlobResponseType, etc
2. Detect the strategy for a given return type
3. Use the strategy to apply headers and such, native http style, check if it supports batching etc
4. On the client read the headers native http and deserialise appropriately
5. Profit

Reply:
yes i agree

how do we get there

Reply:
like, i'd want to have e.g. logic for all diffrent type of content-type (inputs) somewhere and all accept (outputs) somewhere where we have vertical slices of both client + server logic (similar to how jsonl/sse is, but more generic)

the `resolveResponse()`-fn is a mess but unsure exactly how to structure this idea

Reply:
the way we do request content types is pretty clean no?

packages/server/src/unstable-core-do-not-import/http/contentType.ts
packages/client/src/links/internals/contentTypes.ts

Having an array of supported content types with a known interface (strategy pattern) and matching an appropriate strategy is clean and modular, and could be exposed on the public API for extension one day